ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -0,0 +1,163 @@
+import os
+from typing import Optional, Tuple
+
+import click
+
+from ggshield.core.errors import UnexpectedError
+from ggshield.core.git_shell import get_list_commit_SHA
+from ggshield.core.utils import EMPTY_SHA
+
+from .supported_ci import SupportedCI
+
+
+def get_previous_commit_from_ci_env(
+    verbose: bool,
+) -> Tuple[str, SupportedCI]:
+    """
+    Returns the previous HEAD sha of the targeted branch.
+    Returns EMPTY_SHA if there was no commit before.
+    """
+    if os.getenv("GITLAB_CI"):
+        ci_mode = SupportedCI.GITLAB
+        commit_sha = gitlav_previous_commit_sha(verbose)
+
+    elif os.getenv("GITHUB_ACTIONS"):
+        ci_mode = SupportedCI.GITHUB
+        commit_sha = github_previous_commit_sha(verbose)
+
+    elif os.getenv("TRAVIS"):
+        ci_mode = SupportedCI.TRAVIS
+        raise NotImplementedError("Not implemented for Travis.")
+
+    elif os.getenv("JENKINS_HOME") or os.getenv("JENKINS_URL"):
+        ci_mode = SupportedCI.JENKINS
+        raise NotImplementedError("Not implemented for Jenkins.")
+
+    elif os.getenv("CIRCLECI"):
+        ci_mode = SupportedCI.CIRCLECI
+        raise NotImplementedError("Not implemented for CircleCI.")
+
+    elif os.getenv("BITBUCKET_COMMIT"):
+        ci_mode = SupportedCI.BITBUCKET
+        raise NotImplementedError("Not implemented for BitBucket.")
+
+    elif os.getenv("DRONE"):
+        ci_mode = SupportedCI.DRONE
+        raise NotImplementedError("Not implemented for Drone.")
+
+    elif os.getenv("BUILD_BUILDID"):
+        ci_mode = SupportedCI.AZURE
+        raise NotImplementedError("Not implemented for Azure.")
+
+    else:
+        raise UnexpectedError(
+            f"Current CI is not detected or supported."
+            f" Supported CIs: {', '.join([ci.value for ci in SupportedCI])}."
+        )
+    return (commit_sha, ci_mode)
+
+
+def github_previous_commit_sha(verbose: bool) -> str:
+    push_before_sha = github_push_previous_commit_sha()
+    pull_req_base_sha = github_pull_request_previous_commit_sha()
+
+    if verbose:
+        click.echo(
+            f"github_push_before_sha: {push_before_sha}\n"
+            f"github_pull_base_sha: {pull_req_base_sha}\n",
+            err=True,
+        )
+
+    # This one has to be checked before the push one
+    # because this one is only populated in case of PR
+    # whereas push_before_sha can be populated in PR in case of
+    # push force event in a PR
+    if pull_req_base_sha:
+        return pull_req_base_sha
+
+    if push_before_sha:
+        return push_before_sha
+
+    raise UnexpectedError(
+        "Unable to get previous commit. Please submit an issue with the following info:\n"
+        "  Repository URL: <Fill if public>\n"
+        f"github_push_before_sha: {push_before_sha}\n"
+        f"github_pull_base_sha: {pull_req_base_sha}\n"
```

Comment:
`{push_before_sha}` and `{pull_req_base_sha}` are None aren't they?

Reply:
Indeed, or something evaluated to `False`.